h2o-core/src/main/java/water/ChunkAutoBufferWriter.java

```diff
@@ -0,0 +1,89 @@
+package water;
+
+import water.fvec.Chunk;
+import water.fvec.ChunkUtils;
+import water.fvec.Frame;
+import water.parser.BufferedString;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import static water.ExternalFrameUtils.*;
+import static water.ExternalFrameUtils.EXPECTED_STRING;
+
+public final class ChunkAutoBufferWriter implements Closeable  {
+    
+    private final AutoBuffer buffer;
+    
+    public ChunkAutoBufferWriter(OutputStream outputStream) {
+        this.buffer = new AutoBuffer(outputStream, false);
+    }
+
+    public void writeChunk(String frameName, int chunkId, byte[] expectedTypes, int[] selectedColumnIndices) {
```

Comment:
Do you really need to set the vectors types manually ? 

Introducing a function

```java
    private static byte[] vectorTypes(final Frame frame, final int[] vectors){
        final byte[] vectorTypes = new byte[vectors.length];

        for (int i = 0; i < vectors.length; i++) {
            vectorTypes[i] = frame.vec(i).get_type();
        }
        
        return vectorTypes;
    }
```

and then adding one line somewhere at the beginning of this method:

```java
final byte[] expectedTypes = vectorTypes(frame, selectedColumnIndices);
```

and then removing the expectedTypes would do the trick.

Reply:
Oh, I see, those are part of the client request. Do they serve for type checking, or for conversion as well ? If yes, what to do with incovertible types then ?

Do we check the types inside the frame are equal to the types requested ? Something like that.

```java
    private static boolean verifyVectorTypes(final Frame frame, final int[] vectors, final byte[] expectedTypes){

        for (int i = 0; i < vectors.length; i++) {
            if (expectedTypes[i] != frame.vec(vectors[i]).get_type()) return false;
        }
        
        return true;
    }
```